backend/src/index.ts

```diff
@@ -1 +1,60 @@
-console.log("hello world");
+import type { RoomData, BoardState } from "./types";
+
+const http = require("http");
+
+const express = require("express");
+const { Server } = require("socket.io");
+
+const app = express();
+const server = http.createServer(app);
+const io = new Server(server);
+const PORT = process.env.PORT || 3000;
+
+// boards maps room to BoardState (BoardState contains all data in regards to the state of the game and is declared in types.d.ts)
+const boards = new Map<string, BoardState>();
```

Comment:
Does this assign each room a BoardState value? In the BoardState type, what does lock do? I'm also assuming that board  will be the fenstring and players will be a list of names?

Reply:
This assigns each room a BoardState value. The lock variable keeps track of which player's turn it is so that the other player can not make a move when it is not their turn. The board will be the fenstring and players will be a list of usernames.